microscopic-monks/primal/screens/game_screen.py

```diff
@@ -0,0 +1,260 @@
+import random
+
+from primal.engine.screen import Screen
+from primal.engine.camera import OrthographicCamera
+from primal.engine.sprite import Player, ColorSprite
+from primal.engine.world import World
+from primal.engine import keys
+from primal.screens.death_screen import DeathScreen
+
+from primal.gui.health import HealthBar
+from primal.gui.inventory import Inventory
+
+
+class GameScreen(Screen):
+    VP_WIDTH = 1280
+    VP_HEIGHT = 720
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.remove = 1
+
+        self.dividend = 10
+
+        self.clicked_features = dict()
+        self.last_clicked = 0.0
+
+        self.camera = OrthographicCamera(self.canvas, self.VP_WIDTH, self.VP_HEIGHT)
+        self.camera.start_region()
+
+        self.zoom = 1.0
+
+        self.world = World((0, 0))
+        self.world.draw(self.canvas)
+
+        self.player = Player('player.png', (0, 0), (40, 80), 0)
+        self.player.draw(self.canvas)
+
+        self.world.draw_top(self.canvas)
+
+        self.camera.end_region()
+
+        # render gui
+        self.gui_camera = OrthographicCamera(self.canvas, self.VP_WIDTH, self.VP_HEIGHT)
+        self.gui_camera.start_region()
+
+        self.overlay = ColorSprite('overlay.png', (0, 0),
+                                   (self.VP_WIDTH, self.VP_HEIGHT), (1, 1, 1, .3))
+        self.overlay.draw(self.canvas)
+
+        self.health_bar = HealthBar((20, 680), (250, 20), 100.0)
+        self.health_bar.draw(self.canvas)
+
+        self.inventory = Inventory((20, 20))
+        self.inventory.draw(self.canvas)
+        self.gui_camera.end_region()
+        self.timer = 0
+        self.timerValue = 0
+
+        self.update_sected_item()
+
+    def update(self, delta: float):
```

Comment:
The function is really really long, with a lot of business logic that could be factored out. with the use of other methods, docstrings, annotations and the same details to writing short functions that I have seen other places, this would be great.